src/main/java/com/example/demo/tournament/Tournament.java

```diff
@@ -0,0 +1,112 @@
+package com.example.demo.tournament;
+
+import com.example.demo.model.Match;
+import com.example.demo.model.Team;
+import com.example.demo.repository.BaseMatchRepository;
+import com.example.demo.repository.BaseTeamRepository;
+import com.example.demo.repository.TeamRepository;
+import com.opencsv.bean.StatefulBeanToCsv;
+import com.opencsv.bean.StatefulBeanToCsvBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.rmi.UnexpectedException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Supplier;
+
+@Component
+@Value
+@AllArgsConstructor
+public class Tournament {
+    BaseTeamRepository teams;
+    BaseMatchRepository matches;
+
+    public void createMatchesBasedOnListOfTeams() {
+        List<Team> teamList = teams.getList();
+        if (TeamRepository.checkValidCntOfTeams(teamList)) {
+            Random random = new Random();
+            int firstRandomNumber;
+            int secondRandomNumber;
+            int countMatchesAtTheRound = teamList.size() / 2;
+            int sizeOfIndexArray = teamList.size();
+            int cntOfMatches = teamList.size() - 1;
+            int currentRound;
+            int cntOfRounds;
+            List<Integer> indexes = generateIndexList(sizeOfIndexArray);
+
+            Supplier<Integer> returnCntOfRounds = () -> {
```

Comment:
This should be just a private method. If you don't pass this lambda anywhere but just use it here, lambda is redundant